environment.go

```diff
@@ -192,3 +209,56 @@ func (e *envState) Session() db.Session {
 
 	return db.WrapSession(e.session.Copy())
 }
+
+func (e *envState) ClientConfig() *ClientConfig {
+	e.mu.RLock()
+	defer e.mu.RUnlock()
+
+	return e.clientConfig
```

Comment:
I think that you either want this to be a copy of the client config object *or* not a pointer, because otherwise you're not threadsafe. 

It's legit that the settings object suffers from this problem, but that's a legacy problem

Reply:
done